packages/typescript/src/components/TypeFunction.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  childrenArray,
+  findKeyedChild,
+  Prose,
+  Scope,
+  Show,
+} from "@alloy-js/core";
+import { Children } from "@alloy-js/core/jsx-runtime";
+import { getCallSignatureProps } from "../utils.js";
+import { CallSignature, CallSignatureProps } from "./CallSignature.jsx";
+import { Declaration } from "./Declaration.jsx";
+import {
+  FunctionParameters,
+  FunctionTypeParameters,
+  getReturnType,
+  TypeFunctionParameters,
+} from "./FunctionBase.jsx";
+import { JSDoc } from "./JSDoc.jsx";
+import { JSDocParams } from "./JSDocParam.jsx";
+
+/** Define a TypeScript type function
```

Comment:
The docs generator is gonna want this on its own line and the @example wrapped in MD code fence so the syntax is highlighted.